jsonpb/jsonpb_test.go

```diff
@@ -654,15 +674,42 @@ func TestUnmarshalingBadInput(t *testing.T) {
 func TestUnmarshalWithJSONPBUnmarshaler(t *testing.T) {
 	rawJson := `{ "foo": "bar", "baz": [0, 1, 2, 3] }`
 	var msg dynamicMessage
-	err := Unmarshal(strings.NewReader(rawJson), &msg)
-	if err != nil {
+	if err := Unmarshal(strings.NewReader(rawJson), &msg); err != nil {
 		t.Errorf("an unexpected error occurred when parsing into JSONPBUnmarshaler: %v", err)
 	}
 	if msg.rawJson != rawJson {
 		t.Errorf("message contents not set correctly after unmarshalling JSON: got %s, wanted %s", msg.rawJson, rawJson)
 	}
 }
 
+func TestUnmarshalFromAnyWithJSONPBUnmarshaler(t *testing.T) {
+	rawJson := `{ "@type": "blah.com/` + dynamicMessageName + `", "foo": "bar", "baz": [0, 1, 2, 3] }`
+	var a anypb.Any
+	if err := Unmarshal(strings.NewReader(rawJson), &a); err != nil {
+		t.Errorf("an unexpected error occurred when parsing into JSONPBUnmarshaler: %v", err)
+	}
+	var msg ptypes.DynamicAny
```

Comment:
How about constructing an expected Any and compared directly using proto.Equal?

d := &dynamicMessage{rawJson: `{"baz":[0,1,2,3],"foo":"bar"}`}
b, err := proto.Marshal(d)
if err != nil {
    ...
}
want := &anypb.Any{
    TypeURL: "blah.com/` + dynamicMessageName + `",
    Value:  b,
}

if proto.Equal(got, want) ...